### PR TITLE
Sb/alex's refactors for me

### PR DESCRIPTION
**Pull request by SBrandon0432**

### CODE REVIEW

**Review by SBrandon0432 — COMMENTED** [review]
looks good, but we now need a reset on filter settings with the removal of the common denominator of the difficulty settings.

File: src/Components/QuestionList/QuestionFilter/QuestionFilters.tsx [hunk 1]
```diff
@@ -28,7 +28,7 @@ export const QuestionFilters = ({ filterSettings }: Props) => {
                                     type: typeSelect,
                                     name: nameSearch,
                                     hideCorrect: hideCorrect,
-                                    difficulty: difficulty,
+                                    difficulty: difficulty as Difficulty,
```
Comment (CommanderHamster) [inline src/Components/QuestionList/QuestionFilter/QuestionFilters.tsx:31]: If you did your typing correctly, you shouldn't need the `as Difficulty`. That's only needed when you are doing something like `Difficulty` => `any` => `Difficulty`, which you shouldn't be doing

File: src/Components/QuestionList/QuestionList.tsx [hunk 2]
```diff
@@ -23,40 +29,24 @@ export const QuestionList = ({ setShowQuestionList }: any) => {
         type: "Js",
         name: "",
         hideCorrect: false,
-        difficulty: "null",
+        difficulty: "" as Difficulty,
```
Comment (CommanderHamster) [inline src/Components/QuestionList/QuestionList.tsx:32]: `""` is not a valid Difficulty. You need to either default this to `Difficulty.Easy` or create a new item like `Difficulty.None` for the default cause. Up to you.

File: src/Components/Utils/Utils.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { Difficulty, Question } from "../store/types";
+
+export const isQuestionType = (question: Question, type: string) => {
+    return question.tags.includes(type);
```
Comment (CommanderHamster) [inline src/Components/Utils/Utils.tsx:4]: don't forget to check that there is a type filter provided

Comment (CommanderHamster) [reply src/Components/Utils/Utils.tsx:4]: same for name and difficulty

File: src/Components/Utils/Utils.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    difficulty: string
+) => {
+    console.log(question);
+    return question.difficulty.includes(difficulty as Difficulty);
+};
+export const isQuestionHideCorrect = (
+    question: Question,
+    hideCorrect: boolean
+) => {
+    return question.correct === hideCorrect;
```
Comment (CommanderHamster) [inline src/Components/Utils/Utils.tsx:20]: what is this?

File: src/Components/store/index.ts [hunk 5]
```diff
@@ -1,8 +1,8 @@
-import { Question } from "./types";
+import { Difficulty, Question } from "./types";
 
 export let questionBank: Question[] = [
     {
-        difficulty: ["easy", "medium", "hard", "null"],
+        difficulty: "easy" as Difficulty,
```
Comment (CommanderHamster) [inline src/Components/store/index.ts:5]: change these to:
```
difficulty: Difficulty.Easy
```

File: src/Components/Utils/Utils.tsx (hunk 4 above)
Comment (SBrandon0432) [reply src/Components/Utils/Utils.tsx:20]: a check to see if the hide correct and what the answeres sate is set to

**Pull request merged**